relay-server/src/actors/processor.rs

```diff
@@ -2489,6 +2490,62 @@ impl EnvelopeProcessorService {
         Ok(span)
     }
 
+    fn filter_spans(&self, state: &mut ProcessEnvelopeState) {
+        let otel_span_ingestion_enabled =
+            state.project_state.has_feature(Feature::OtelSpanIngestion);
+        state.managed_envelope.retain_items(|item| match item.ty() {
+            ItemType::OtelSpan if !otel_span_ingestion_enabled => {
+                relay_log::warn!("dropping span because feature is disabled");
+                ItemAction::DropSilently
+            }
+            _ => ItemAction::Keep,
+        });
+    }
+
+    #[cfg(feature = "processing")]
+    fn process_spans(&self, state: &mut ProcessEnvelopeState) {
+        let mut items: Vec<Item> = Vec::new();
+        state.managed_envelope.retain_items(|item| match item.ty() {
+            ItemType::OtelSpan => {
+                let span: relay_spans::Span = match serde_json::from_slice(&item.payload()) {
```

Comment:
nit: Could use [`let .. else`](https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html) here.

Reply:
Does it mean I need to drop the error logging or is there a way to do it with error logging as well?

Reply:
Oh right, you can't get the error info that way. Ignore me.